### ISSUE TITLE
`tokio_threadpool::blocking` is incompatible with `current_thread::Runtime`

### ISSUE THREAD

**Issue by danburkert** [tokio-rs/tokio#432]
The following test fails:

```rust
#[test]
fn current_thread_blocking() {
    let _ = env_logger::init();

    let mut runtime = tokio::runtime::current_thread::Runtime::new().unwrap();

    runtime.block_on(lazy(|| {
        tokio_threadpool::blocking(|| {
        })
    })).unwrap();
}
```
with `thread 'current_thread_blocking' panicked at 'called `Result::unwrap()` on an `Err` value: BlockingError { _p: () }', libcore/result.rs:945:5`

But it succeeds if the multi-threaded runtime is used instead.  I suspect this means the entire `fs` crate is unusable with the current thread runtime, however I haven't tested that.

**Comment by carllerche** [comment]
As discussed on Gitter, there probably is a way to implement something that works on other runtimes if the closure is `Send + 'static`.

If called from a threadpool, then `threadpool::blocking` is used. Otherwise, dispatch the closure to a threadpool and run it there.

**Comment by kpp** [comment]
```
impl fmt::Display for BlockingError {
    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
        write!(fmt, "{}", self.description())
    }
}

impl Error for BlockingError {
    fn description(&self) -> &str {
        "`blocking` annotation used from outside the context of a thread pool"
    }
}
```

And `derive Debug`.

Why do we see `an \`Err\` value: BlockingError { _p: () }`?

**Comment by kpp** [comment]
@danburkert shall we close your issue after #451 ?

**Comment by danburkert** [comment]
@kpp #451 just improved the error message, it's still not possible to use the blocking API with the current thread executor.

**Comment by kpp** [comment]
I agree. My goal was to reduce the number of WTFs.

**Comment by carllerche** [comment]
Closing in favor of #588
